homeassistant/components/weather/smhi.py

```diff
@@ -73,11 +73,11 @@
                             config_entries) -> bool:
     """Add a weather entity from map location."""
     location = config_entry.data
-    name = location[CONF_NAME]
+    name = slugify(location[CONF_NAME])
 
     session = aiohttp_client.async_get_clientsession(hass)
 
-    entity = SmhiWeather(name, location[CONF_LATITUDE],
+    entity = SmhiWeather(location[CONF_NAME], location[CONF_LATITUDE],
                          location[CONF_LONGITUDE],
                          session=session)
     entity.entity_id = ENTITY_ID_SENSOR_FORMAT.format(name)
```

Comment:
The solution is to just not assign your own entity. That's not necessary at all.

Reply:
Remove the line 83 basically?

Reply:
ugh okay, we can keep it. It's just silly that we're duplicating logic :(

Reply:
Agreed, my fault.. I should have done my home work better.